src/legacy/core_plugins/console/server/request.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import http from 'http';
+import https from 'https';
+import net from 'net';
+import stream from 'stream';
+import Boom from 'boom';
+import { URL } from 'url';
+
+interface Args {
+  method: string;
+  agent: http.Agent;
+  uri: URL;
+  payload: stream.Stream;
+  timeout: number;
+  headers: http.OutgoingHttpHeaders;
+}
+
+// We use a modified version of Hapi's Wreck because  Hapi, Axios, and Superagent don't support GET requests
+// with bodies, but ES APIs do. Similarly with DELETE requests with bodies. Another library, `request`
+// diverged too much from current behaviour.
+export const sendRequest = ({ method, headers, agent, uri, timeout, payload }: Args) => {
+  const client = uri.protocol === 'https:' ? https : http;
+  let resolved = false;
+
+  let resolve: any;
+  let reject: any;
+  const reqPromise = new Promise<http.ServerResponse>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+
+  const req = client.request({
+    method: method.toUpperCase(),
+    host: uri.hostname,
+    port: uri.port !== '' ? Number(uri.port) : undefined,
+    protocol: uri.protocol,
+    path: `${uri.pathname}${uri.search || ''}`,
+    headers: {
+      ...headers,
+      'content-type': 'application/json',
+      'transfer-encoding': 'chunked',
```

Comment:
👍

Reply:
i don't feel very strongly about this, but think it's fine.  the query may have a lot of data, but i would expect the ui to lock up before that's an issue.  ace used to have a max of 65k lines but i'm a little out of date